### PR TITLE
[chttp2] Remove absl::optional for TaskHandle

### PR DESCRIPTION
**Pull request by ananda1066**

### CODE REVIEW

File: src/core/ext/transport/chttp2/transport/chttp2_transport.cc [hunk 1]
```diff
@@ -812,28 +812,38 @@ static void close_transport_locked(grpc_chttp2_transport* t,
           t->settings_ack_watchdog,
           grpc_event_engine::experimental::EventEngine::TaskHandle::kInvalid));
     }
-    if (t->delayed_ping_timer_handle.has_value()) {
-      if (t->event_engine->Cancel(*t->delayed_ping_timer_handle)) {
-        t->delayed_ping_timer_handle.reset();
+    if (t->delayed_ping_timer_handle !=
```
Comment (drfloob) [inline src/core/ext/transport/chttp2/transport/chttp2_transport.cc:815]: Drive-by comment: could be more readable with something like:

```
// Added somewhere file-scoped, up above
using grpc_event_engine::experimental::EventEngine::TaskHandle;
...
if (t->delayed_ping_timer_handle != TaskHandle::kInvalid &&
    t->event_engine->Cancel(t->delayed_ping_timer_handle)) {
  t->delayed_ping_timer_handle = TaskHandle::kInvalid;
}
```

Comment (ananda1066) [reply src/core/ext/transport/chttp2/transport/chttp2_transport.cc]: Done. @drfloob adding you as reviewer since I take this as volunteering :)

File: src/core/ext/transport/chttp2/transport/chttp2_transport.cc [hunk 2]
```diff
@@ -803,41 +806,42 @@ static void close_transport_locked(grpc_chttp2_transport* t,
...
+      t->next_bdp_ping_timer_handle = TaskHandle::kInvalid;
     }
     switch (t->keepalive_state) {
       case GRPC_CHTTP2_KEEPALIVE_STATE_WAITING:
-        if (t->keepalive_ping_timer_handle.has_value()) {
-          if (t->event_engine->Cancel(*t->keepalive_ping_timer_handle)) {
-            t->keepalive_ping_timer_handle.reset();
-          }
+        if (t->keepalive_ping_timer_handle !=
+                grpc_event_engine::experimental::EventEngine::TaskHandle::
```
Comment (drfloob) [inline src/core/ext/transport/chttp2/transport/chttp2_transport.cc:831]: This can just be `TaskHandle::kInvalid`

File: src/core/ext/transport/chttp2/transport/chttp2_transport.cc [hunk 3]
```diff
@@ -803,41 +806,42 @@ static void close_transport_locked(grpc_chttp2_transport* t,
...
+                    kInvalid &&
+            t->event_engine->Cancel(t->keepalive_ping_timer_handle)) {
+          t->keepalive_ping_timer_handle = grpc_event_engine::experimental::
```
Comment (drfloob) [inline src/core/ext/transport/chttp2/transport/chttp2_transport.cc:834]: This can just be `TaskHandle::kInvalid`

File: src/core/ext/transport/chttp2/transport/chttp2_transport.cc [hunk 4]
```diff
@@ -803,41 +806,42 @@ static void close_transport_locked(grpc_chttp2_transport* t,
...
+              EventEngine::TaskHandle::kInvalid;
         }
         break;
       case GRPC_CHTTP2_KEEPALIVE_STATE_PINGING:
-        if (t->keepalive_ping_timer_handle.has_value()) {
-          if (t->event_engine->Cancel(*t->keepalive_ping_timer_handle)) {
-            t->keepalive_ping_timer_handle.reset();
-          }
+        if (t->keepalive_ping_timer_handle !=
+                grpc_event_engine::experimental::EventEngine::TaskHandle::
```
Comment (drfloob) [inline src/core/ext/transport/chttp2/transport/chttp2_transport.cc:840]: This can just be `TaskHandle::kInvalid`

File: src/core/ext/transport/chttp2/transport/chttp2_transport.cc [hunk 5]
```diff
@@ -803,41 +806,42 @@ static void close_transport_locked(grpc_chttp2_transport* t,
...
+                    kInvalid &&
+            t->event_engine->Cancel(t->keepalive_ping_timer_handle)) {
+          t->keepalive_ping_timer_handle = grpc_event_engine::experimental::
```
Comment (drfloob) [inline src/core/ext/transport/chttp2/transport/chttp2_transport.cc:843]: This can just be `TaskHandle::kInvalid`

**Review by drfloob — APPROVED** [review]
Looks good, just a few more small corrections.
